src/main/java/nl/knaw/dans/migration/core/ExpectedLoader.java

```diff
@@ -30,21 +31,24 @@ public class ExpectedLoader {
 
   private final ExpectedFileDAO expectedFileDAO;
   private final ExpectedDatasetDAO expectedDatasetDAO;
+  final InputDatasetDAO inputDatasetDAO;
   private final Map<String, String> userToEmail;
   private final Map<String, String> licensesUrlToName;
 
-  public ExpectedLoader(ExpectedFileDAO expectedFileDAO, ExpectedDatasetDAO expectedDatasetDAO, File configDir) {
+  public ExpectedLoader(ExpectedFileDAO expectedFileDAO, ExpectedDatasetDAO expectedDatasetDAO, InputDatasetDAO inputDatasetDAO, File configDir) {
     this.expectedFileDAO = expectedFileDAO;
     this.expectedDatasetDAO = expectedDatasetDAO;
+    this.inputDatasetDAO = inputDatasetDAO;
     this.userToEmail = Mapping.load(new File(configDir + "/easy-users.csv"), "UID", "email");
     this.licensesUrlToName = Mapping.load(new File(configDir + "/licenses.csv"),"url","name");
   }
 
-  public void deleteByDoi(String doi, Mode mode) {
+  public void deleteByDoi(String doi, Mode mode, String source) {
     if (mode.doFiles())
       expectedFileDAO.deleteByDoi(doi);
     if (mode.doDatasets())
       expectedDatasetDAO.deleteByDoi(doi);
+    inputDatasetDAO.deleteByDoi(doi, source);
```

Comment:
Don't remember why at some point in time I wanted this last line only for load-from-vault.